src/gui/accountsettings.cpp

```diff
@@ -62,6 +62,28 @@
 
 namespace {
 constexpr auto propertyFolderInfo = "folderInfo";
+
+bool showEnableE2eeWithVirtualFilesWarningDialog()
+{
+    QMessageBox e2eeWithVirtualFilesWarningMsgBox;
+    e2eeWithVirtualFilesWarningMsgBox.setText("End-to-End Encryption with Virtual Files");
+    e2eeWithVirtualFilesWarningMsgBox.setInformativeText("You seem to have the Virtual Files feature enabled on this folder. At "
+                                                         " the moment, it is not possible to implicit downloading virtual files that are "
+                                                         "End-to-End encrypted. To get the best experience with Virtual Files and"
+                                                         " End-to-End Encryption, make sure the encrypted folder is marked with"
+                                                         " \"Make always available locally\".");
+    e2eeWithVirtualFilesWarningMsgBox.setIcon(QMessageBox::Warning);
+    const auto dontEncryptButton = e2eeWithVirtualFilesWarningMsgBox.addButton("Don't encrypt this folder", QMessageBox::AcceptRole);
```

Comment:
It's more common to override text of say... ok and cancel and then rely on if it was accepted or not. This avoids the unreachable and you can return a simple condition over the exec() value like for a regular dialog.

Reply:
Is it now better?